web_src/js/components/DiffFileTreeItem.vue

```diff
@@ -32,14 +32,19 @@
         <span class="gt-ellipsis">{{ item.name }}</span>
       </div>
       <div v-show="!collapsed">
-        <DiffFileTreeItem v-for="childItem in item.children" :key="childItem.name" :item="childItem" class="list" :selected-file="selectedFile"/>
+        <DiffFileTreeItem v-for="childItem in item.children" :key="childItem.name" :item="childItem" class="list"/>
       </div>
     </div>
   </div>
 </template>
 
 <script>
 import {SvgIcon} from '../svg.js';
+import {reactive} from 'vue';
+
+export const store = reactive({
+  selectedItem: '',
+});
```

Comment:
Maybe make a separate `stores.js` file and import that into both components? I think we should generally avoid exporting from vue SFCs, except for testing.

Reply:
> avoid exporting from vue SFCs

What's the idea / best practice behind it?

----

I put the `store` inside `DiffFileTreeItem`, because in `DiffFileTree.vue`, the `DiffFileTreeItem` was already imported there, there is a direct dependency between them. Now it is `import DiffFileTreeItem, {store} from './DiffFileTreeItem.vue'` , I don't see any problem.

Reply:
1. I just find it cleaner if all stores/states are in their own file and can be managed at once.
1. Imho, components should only import from each another if they render said component.
1. I also recall one specific issue with React components where hot module replacement does not work if component files export non-components. It may be that Vue HMR has similar restrictions, should we ever decide to use it.

Reply:
Done in f3b7776cf97f66d395205e070f5c93a8737592d8

Reply:
I didn't meant you to create one file per store, but one file for all stores, exported by name.

Reply:
I don't see why we should put all stores together since they are totally unrelated.

Reply:
I don't like to put unrelated code in a single file because they may cause unnecessary conflicts when a project is being actively developed.

I am not sure I can make a dir/file layout you like, so feel free to edit on this PR directly.

Reply:
I do think it's common practice to put all stores together in one file, at least in React. In fact, the [zustand](https://github.com/pmndrs/zustand) forces one to do that. But again, I'm not an authorative source of best practices for Vue.

Reply:
I'll refactor to my proposed layout later.